### ISSUE TITLE
Upgrade to GMavenPlus 1.6.3

### ISSUE THREAD

**Issue by snicoll** [spring-io/initializr#875]
Hi, this is a `first-timers-only` issue. This means we've worked to make it more legible to folks who either haven't contributed to our codebase before, or even folks who haven't contributed to open source before.

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

If you have contributed before, consider leaving this one for someone new, and looking through our general `ideal-for-contribution` issues. Thanks!

## Background
Project generation works by applying a set of `ProjectContributor` implementations on a model. Each model is customized according to various parameters. For instance, our Maven build is customized specifically for Groovy using `io.spring.initializr.generator.spring.code.groovy.GroovyMavenBuildCustomizer`. 

## Problem
A Groovy-based build with Maven uses the [GMavenPlus plugin](https://github.com/groovy/GMavenPlus). The project uses version `1.5` but version `1.6.3` just got released so we should make sure our integration uses it.

It's not a simple version switch though as several goals [were renamed to be more consistent](https://github.com/groovy/GMavenPlus/issues/31).

## Solution
The version of the plugin to use should be changed from `1.5` to `1.6.3` in `GroovyMavenBuildCustomizer`. Two goals have to be renamed as well:

* `testGenerateStubs` to `generateTestStubs`
* `testCompile` to `compileTests`

There is a test in `GroovyMavenBuildCustomizerTests` that asserts the model has been properly updated which will need to be changed.

## Steps to Fix 
 
- [x] Claim this issue with a comment below and ask any clarifying questions you need
- [ ] Set up a repository locally following the [Contributing Guidelines](https://github.com/spring-io/initializr/blob/master/CONTRIBUTING.adoc)
- [ ] Try to fix the issue following the steps above
- [ ] Commit your changes and start a pull request.

**Comment by sundarraj4u** [comment]
I can work on this one, please let me know

**Comment by snicoll** [comment]
Great! Thank you. Please let us know if you have any questions.

**Comment by roussi** [comment]
Can I work on it in parallel of other new contributors ?

**Comment by snicoll** [comment]
@roussi thanks for the offer. You're free to give it a try on your own if you want but please give enough time to @sundarraj4u to submit something. If it turns out that @sundarraj4u can no longer work on it, I'll ping you for sure.

**Comment by roussi** [comment]
@snicoll thanks

**Comment by sundarraj4u** [comment]
@snicoll i send you a pull request for this issue.

**Comment by snicoll** [comment]
@sundarraj4u thanks for working on this. FYI, there is no need to add a comment here, all watchers already received a notification when you opened the pull request.

**Comment by snicoll** [comment]
Closing in favour of PR #878
